util/include/VirtualFunctionTemplate.h

```diff
@@ -54,8 +54,9 @@
   template<typename T>                                                                                                 \
   class functionName##_functionSignature : public boost::function<__VA_ARGS__> {                                       \
    public:                                                                                                             \
-    boost::function<__VA_ARGS__>& operator=(const boost::function<__VA_ARGS__>& rhs) {                                 \
-      return boost::function<__VA_ARGS__>::operator=(rhs);                                                             \
+    functionName##_functionSignature& operator=(const boost::function<__VA_ARGS__>& rhs) {                             \
```

Comment:
change return type

Reply:
I have no clue what this is supposed to do